src/components/graph/graph.helper.js

```diff
@@ -469,7 +469,144 @@ function normalize(vector) {
   return norm === 0 ? vector : { x: vector.x / norm, y: vector.y / norm };
 }
 
-const SYMBOLS_WITH_OPTIMIZED_POSITIONING = new Set([CONST.SYMBOLS.CIRCLE]);
+/**
+ * Calculate the length of a vector, from the center of a rectangle,
+ * to one of it's edges.
+ * This calculation is taken from https://stackoverflow.com/a/3197924.
+ *
+ * @param {Object.<string, number>} RectangleCoords, The coords of the left-top vertex, and the right-bottom vertex.
+ * @param {Object.<string, number>} VectorOriginCoords The center of the rectangle coords.
+ * @param {Object.<string, number>} directionVector a 2D vector with x and y components
+ */
+function calcRectangleVectorLengthFromCoords({ x1, y1, x2, y2 }, { px, py }, directionVector) {
```

Comment:
Thi seems like redundant information since you can compute 
``` js
let px = (x1 + x2)/2; 
let py = (y1 + y2)/2
```

Maybe remove?

Reply:
If we assume that `px` and `py` represent the center of the **Square**, `x` and `y` match our needs, in this library `x` and `y` represent the center of a symbol (i.e. square, circle, wye), and not it's top left point.

Reply:
If I am only looking at the data you are inputing into your method : (graph.coords.js, line 90): 
``` javascript
const leftSquareX = x - edgeSize / 2;
const topSquareY = y - edgeSize / 2;

{ x1: leftSquareX, y1: topSquareY, x2: leftSquareX + edgeSize, y2: topSquareY + edgeSize }
```

So here 
`(x1 + x2)/2 = (leftSquareX + leftSquareX + edgeSize)/2 = (2 * x - edgeSize + edgeSize)/2 = x`
and
`(y1 + y2)/2 = (topSquareY + topSquareY+ edgeSize)/2 = (2 * y - edgeSize + edgeSize)/2 = y`

And it seems to be the same calculation when you call the method the other time. 
Strictly from a parameters/data point of view, I feel like it is redundant information (since you are only calling this method for square/rectangle)
Unless you plan on calling it somewhere else?